component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/elk/ELKMetricEventBuilder.java

```diff
@@ -32,12 +35,26 @@
  * Event builder for log Metric Reporter
  */
 public class ELKMetricEventBuilder extends AbstractMetricEventBuilder {
-    private Map<String, Object> eventMap = new HashMap<>();
+
+    protected Map<String, Class> requiredAttributes;
+    private Map<String, Object> eventMap;
     private Boolean isBuilt = false;
 
+
+    public ELKMetricEventBuilder(){
+        requiredAttributes = GenericInputValidator.getInstance().getEventProperties(MetricSchema.RESPONSE);
```

Comment:
Why are we introducing the concept of required attributed to the ELK builder?

Reply:
Nuwan suggested me to use similar filtering functionality in the ELK scenario as well.

Reply:
Yes it does Tishan.

Reply:
Please talk with @AnuGayan to verify that ELK upstream doesnt get break due to us adding extra attribute

Reply:
I will do Tishan.
From my side I did test it and works accordingly.